localize/commands.py

```diff
@@ -64,12 +67,11 @@ def push(conf):
     content={ 'content': file }
 
     # Use the file extension to guess the language and format
-    base=os.path.basename(source['file'])
-    language=os.path.splitext(base)[0]
-    format=os.path.splitext(base)[1]
+    base = os.path.basename(source['file'])
+    language, format = check_and_return_lang_format(base, 'push')
     data={
       'language': language,
-      'format': format.replace('.','').upper()
+      'format': format.replace('yml','yaml').replace('.','').upper()
```

Comment:
No, after the first replace will return string, and second will replace dot by '' and will return string to upper function.

But good catch. After i created `check_and_return_lang_format` the second replace became redundant.